dotnet/src/SemanticKernel/SkillDefinition/InlineFunctionsDefinitionExtension.cs

```diff
@@ -49,19 +51,30 @@ public static class InlineFunctionsDefinitionExtension
     {
         functionName ??= RandomFunctionName();
 
-        var config = new PromptTemplateConfig
+        JsonArray stopSequencesArray = new();
+
+        if (stopSequences != null)
         {
-            Description = description ?? "Generic function, unknown purpose",
-            Type = "completion",
-            Completion = new PromptTemplateConfig.CompletionConfig
+            foreach (var stopSequence in stopSequences)
             {
-                Temperature = temperature,
-                TopP = topP,
-                PresencePenalty = presencePenalty,
-                FrequencyPenalty = frequencyPenalty,
-                MaxTokens = maxTokens,
-                StopSequences = stopSequences?.ToList() ?? new List<string>()
+                stopSequencesArray.Add(JsonValue.Create<string>(stopSequence));
             }
+        }
+
+        var serviceSettings = new JsonObject
+        {
+            ["temperature"] = temperature,
```

Comment:
this is one place where we can see the problem with JsonObject:
* Keys are hardcoded
* There is no way to know which keys are required - normally one would just use IntelliSense
* there is no validation, each value is an object

Reply:
We can still use serialization/deserialization and work with strongly typed models on AI services level (like in `ClientBase.cs` class of OpenAI connector). On AI service level we still perform validation.